src/java.desktop/windows/classes/sun/awt/windows/WInputMethod.java

```diff
@@ -151,9 +162,7 @@ public synchronized void setInputMethodContext(InputMethodContext context) {
 
     @Override
     public void dispose() {
-        // Due to a memory management problem in Windows 98, we should retain
-        // the native input context until this object is finalized. So do
-        // nothing here.
+        disposerRecord.dispose();
```

Comment:
The native dispose will call `ImmDestroyContext` which is specified as:
> However, before calling ImmDestroyContext, the application must remove the input context from any association with windows in the thread by using the [ImmAssociateContext](https://learn.microsoft.com/en-us/windows/desktop/api/imm/nf-imm-immassociatecontext) function.

I did not check the code, but maybe we are calling `ImmAssociateContext` in the wrong moment and have to postpone this "dispose" until all the object is really unused?

Reply:
This seems to be done by a native method WInputMethod.disableNativeIME(..) - which sets the IME for the window to NULL.  The native method is called from various places in WInputMethod.
I can't see that dispose or finalize would do anything to ensure that is called, or introduce any timing issues either way. So if there's a problem it is un-related.